src/components/Ido/Ido.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { Grid, Typography, Box, CardMedia } from '@material-ui/core'
+import { SwapToken } from '@components/Swap/Swap'
+
+import useStyle from './style'
+import AmountInput from '@components/Inputs/ExchangeAmountInput/ExchangeAmountInput'
+import AnimatedButton from '@components/AnimatedButton/AnimatedButton'
+import icons from '@static/icons'
+import classNames from 'classnames'
+import { printBN } from '@consts/utils'
+
+interface IdoInterface {
+  xBtc: string
+  xEth: string
+  sol: string
+  usd: string
+  xUsd: string
+  header: string
+  buttonHeader: string
+  tokens: SwapToken[]
+}
+
+const Ido: React.FC<IdoInterface> = ({
+  header,
+  xUsd,
+  usd,
+  sol,
+  xEth,
+  xBtc,
+  buttonHeader,
+  tokens
+}) => {
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [tokenIndex, setTokenIndex] = React.useState<number | null>(tokens.length ? 0 : null)
+
+  const classes = useStyle()
+
+  return (
+    <Grid className={classes.idoWrapper}>
+      <Grid container className={classes.header}>
+        <Typography component='h1'>IDO</Typography>
+      </Grid>
+      <Grid container className={classes.root} direction='column'>
+        <Box className={classes.depositHeader}>
+          <Typography component='h1'>{header}</Typography>
+        </Box>
+        <Grid className={classes.AmountInputContainer}>
+          <AmountInput
```

Comment:
one last thing: maybe use DepositAmountInput instead; there will be only one token to deposit, probably SOL or mSOL

Reply:
done